### PR TITLE
Issue/98

### PR DESCRIPTION
**Pull request by gytis-ivaskevicius**
Should improve #98. 
Added a couple of extra checks which hopefully will catch at least a few edge cases. Unless somebody else has any suggestions - there is nothing else one may do for that particular issue.

Also for the record, I have not tested that function much, maybe someone could confirm that it works as expected?

### CODE REVIEW

**Review by Pacman99 — APPROVED** [review]
Yeah this is a pretty effective solution for this, thanks!
But I'm really not a huge fan of the complexity of this solution, and the growing complexity of exportOverlays, which I know is largely on me.
I'm starting to feel like automatic exporting isn't the best idea. I quite like how clean the selective exporting of modules in digga has started to become.

File: lib/exportOverlays.nix [hunk 1]
```diff
@@ -44,27 +44,44 @@ let
...
         foldl'
         head
+        isAttrs
+        isFunction
         listToAttrs
         mapAttrs
+        tryEval
         ;
+
+      # Hopefully will fix a couple of edge cases. Even tho `lib` is defined as API here - it is not. Do not use it.
```
Comment (Pacman99) [inline lib/exportOverlays.nix:59]: Is this note meant for developers of exportOverlays? or users of `fakePkgs`

**Comment by Pacman99** [comment]
This only somewhat deals with the issue, the larger culprit will likely be `exportPackages`

File: lib/exportOverlays.nix (hunk 1 above)
Comment (gytis-ivaskevicius) [reply lib/exportOverlays.nix:59]: I guess users if they decide to check the source

**Pull request merged**